shared/src/commonMain/kotlin/org/hyperskill/app/track/presentation/TrackReducer.kt

```diff
@@ -63,27 +73,24 @@ class TrackReducer(
                     GamificationToolbarFeature.Message.PullToRefresh(GamificationToolbarScreen.TRACK)
                 )
 
-                state.copy(trackState = trackState, toolbarState = toolbarState) to trackActions + toolbarActions
+                val (topicsToDiscoverNextState, topicsToDiscoverNextActions) = reduceTopicsToDiscoverNextMessage(
+                    state.topicsToDiscoverNextState,
+                    TopicsToDiscoverNextFeature.Message.PullToRefresh
+                )
+
+                state.copy(
+                    trackState = trackState,
+                    toolbarState = toolbarState,
+                    topicsToDiscoverNextState = topicsToDiscoverNextState
+                ) to trackActions + toolbarActions + topicsToDiscoverNextActions
             }
             // Click Messages
             is Message.TopicToDiscoverNextClicked -> {
-                val targetTheoryId = (state.trackState as? TrackState.Content)
-                    ?.topicsToDiscoverNext?.firstOrNull { it.id == message.topicId }
-                    ?.theoryId
-
-                if (targetTheoryId != null) {
-                    state to setOf(
-                        Action.ViewAction.NavigateTo.StepScreen(targetTheoryId),
-                        Action.LogAnalyticEvent(
-                            TrackClickedTopicToDiscoverNextHyperskillAnalyticEvent(
-                                topicId = message.topicId,
-                                theoryId = targetTheoryId
-                            )
-                        )
-                    )
-                } else {
-                    null
-                }
+                val (topicsToDiscoverNextState, topicsToDiscoverNextActions) = reduceTopicsToDiscoverNextMessage(
+                    state.topicsToDiscoverNextState,
+                    TopicsToDiscoverNextFeature.Message.TopicToDiscoverNextClicked(message.topicId)
+                )
+                state.copy(topicsToDiscoverNextState = topicsToDiscoverNextState) to topicsToDiscoverNextActions
             }
```

Comment:
Why not to replace it with a platform side `onNewMessage(TrackFeature.TopicToDiscoverNextMessage(...))` call?